include/renderer_vk/vk_debug.hpp

```diff
@@ -0,0 +1,48 @@
+#pragma once
+
+#include <span>
+#include <type_traits>
+#include <utility>
+
+#include "vulkan_api.hpp"
+
+namespace Vulkan {
+
+	VKAPI_ATTR VkBool32 VKAPI_CALL debugMessageCallback(
+		VkDebugUtilsMessageSeverityFlagBitsEXT messageSeverity, VkDebugUtilsMessageTypeFlagsEXT messageType,
+		const VkDebugUtilsMessengerCallbackDataEXT* callbackData, void* userData
+	);
+
+	void setObjectName(vk::Device device, vk::ObjectType objectType, const void* objectHandle, const char* format, ...);
+
+	template <typename T, typename = std::enable_if_t<vk::isVulkanHandleType<T>::value == true>, typename... ArgsT>
```

Comment:
This signature is legitimately fascinating to look qt

Reply:
It's basically a compile-time way to derive a `vk::ObjectType`-enum based on the `vk::blah` that gets passed in, and will yell at you at compile-time if you try to give it something weird.